packages/vscode-extension/lib/wrapper.js

```diff
@@ -69,6 +69,126 @@ function useAgentListener(agent, eventName, listener, deps = []) {
   }, [agent, ...deps]);
 }
 
+function obtainGetInspectorDataForInstance() {
+  const renderers = Array.from(window.__REACT_DEVTOOLS_GLOBAL_HOOK__?.renderers?.values());
+  if (!renderers) {
+    return undefined;
+  }
+  for (const renderer of renderers) {
+    if (renderer.rendererConfig?.getInspectorDataForInstance) {
+      return renderer.rendererConfig.getInspectorDataForInstance;
+    }
+  }
+  return undefined;
+};
+
+function createStackElement(
+  frame, name, source
+) {
+  return {
+    componentName: name,
+    source: {
+      fileName: source.fileName,
+      line0Based: source.lineNumber - 1,
+      column0Based: source.columnNumber - 1,
+    },
+    frame,
+  };
+};
+
+// Returns an array of promises which resolve to elements of the components hierarchy stack.
+function traverseComponentsTreeUp(startNode) {
+  const getInspectorDataForInstance = obtainGetInspectorDataForInstance();
+
+  if (!getInspectorDataForInstance) {
+    console.warn("Failed to obtain getInspectorDataForInstance renderer method.");
+    return [];
+  }
+
+  const stackPromises = [];
+  let node = startNode;
+
+  // Optimization: we break after reaching fiber node corresponding to OffscreenComponent (with tag 22).
+  // https://github.com/facebook/react/blob/c3570b158d087eb4e3ee5748c4bd9360045c8a26/packages/react-reconciler/src/ReactWorkTags.js#L62
+  while (node && node.tag !== 22) {
+    const data = getInspectorDataForInstance(node);
+
+    if (data.hierarchy && data.hierarchy.length > 0) {
+      stackPromises.push(new Promise((resolve, reject) => {
+        const item = data.hierarchy[data.hierarchy.length - 1];
+        const inspectorData = item.getInspectorData(findNodeHandle);
+        
+        try {
+          inspectorData.measure((_x, _y, viewWidth, viewHeight, pageX, pageY) => {
+            const stackElementFrame = {
+              x: pageX,
+              y: pageY,
+              width: viewWidth,
+              height: viewHeight
+            };
+
+            stackElement = (inspectorData.source) ? 
```

Comment:
```suggestion
            const stackElement = (inspectorData.source) ? 
```

Reply:
Also, it appears like we can tell whether source is set before we call measure, so maybe we can skip calling measure altogether in such a case?

Reply:
added an if statement before measure to check if source is absent and resolve the promise early in such case